src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -452,6 +452,23 @@ const JupyterFrameManager = ({ onClose, frameRef }) => {
   return null
 }
 
+const PeriodicCookieSetter = ({ namespace, clusterName }) => {
+  const signal = useCancellation()
+
+  const periodicallySetCookie = withErrorReporting('Error setting cookie', async () => {
```

Comment:
With the error handler here, a single failed call would cause the loop to exit. Perhaps we should catch the error but continue looping, in case it was just a network blip. Also, we should think carefully about whether we actually want to report or ignore the error, since a string of failures might generate a lot of notifications.